plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XWorkspaceBuilder.java

```diff
@@ -53,6 +53,31 @@
 @SuppressWarnings({ "hiding", "restriction" })
 public class XWorkspaceBuilder {
 
+	private static class AffectedResourcesRecordingFactory implements IBuildRequestFactory {
+		private final IBuildRequestFactory delegate;
+		private final Set<URI> affected;
+
+		private AffectedResourcesRecordingFactory(IBuildRequestFactory delegate) {
+			this.delegate = delegate;
+			this.affected = new HashSet<>();
+		}
+
+		@Override
+		public XBuildRequest getBuildRequest(String projectName, Set<URI> changedFiles,
+				Set<URI> deletedFiles, List<Delta> externalDeltas) {
+			XBuildRequest result = delegate.getBuildRequest(projectName, changedFiles, deletedFiles, externalDeltas);
+			result.setAffectedListener(affected::add);
```

Comment:
I wonder if the anonymous class created here at `affected::add` might be a good place for breakpoints or temporary logging during debugging. In that case it would be good to write this more lengthy.

Reply:
Do you think that `XBuildRequest. afterDetectedAsAffected` is a fair location for a breakpoint?

Reply:
Maybe a bit more difficult to find when working on the XWorkspaceBuilder, but it should do the trick.